### PR TITLE
Authorization with can can can

### PR DESCRIPTION
**Pull request by Beardless-sheik**
In this pull request: 
- CanCanCan ruby gem installed to the application gemfile.
- "Role" attribute was created on User table using rails migration. 
- Ability file is configured to allow authorization based on project guidelines. 
- Post and Comment deletion implemented using authorization rules from CanCan Authorization Ability config file.

### CODE REVIEW

**Review by FrederickMih — COMMENTED** [review]
Hi @Beardless-sheik  , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

#### This is a collaborative project. Kindly ensure there are commits from both partners.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -23,14 +23,24 @@ def create
...
       format.html do
         if new_post.save
-          redirect_to post_path(params[:user_id]), notice: 'Post was successfully created.'
+          redirect_to user_posts_path(params[:user_id]), notice: 'Post was successfully created.'
         else
           render :new, alert: 'Error occurred, please try again. Post not saved'
         end
       end
     end
   end
```
Comment (Mre55) [inline app/controllers/posts_controller.rb:34]: Kindly, use `CanCanCan` for Post authorization here in `Post Controller`. To do that you can include something like this at the top between lines 1 and 2:
```
load_and_authorize_resource
```

You can read more [here](https://www.rubydoc.info/github/CanCanCommunity/cancancan/CanCan%2FControllerAdditions%2FClassMethods:load_and_authorize_resource).

**Review by Mre55 — CHANGES_REQUESTED** [review]
**Hi @Beardless-sheik and @Abenezer-Tilahun** 

You have done a great job so far! 💪 👏 
You still need to work on a few issues before moving on to the next project, but you're almost there!

## To highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- The README is well documented ✅ 
- Linters are working properly ✅ 

## Required Changes ♻️🛠️ 

- [ ] Kindly, the person with the less commit will have to do this change in order to keep the commit balance.

_Check the comments under the review._




## Optional suggestions 🟡 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Beardless-sheik) [reply app/controllers/posts_controller.rb:34]: Thanks for the suggestion @Mre55 . Will implement

**Review by t-yanick — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**